src/PowerShellEditorServices/Session/Host/EditorServicesPSHostUserInterface.cs

```diff
@@ -32,6 +32,7 @@ public abstract class EditorServicesPSHostUserInterface :
     {
         #region Private Fields
 
+        private readonly HashSet<ProgressKey> currentProgressMessages = new HashSet<ProgressKey>();
```

Comment:
thread safety pls 😄 [ConcurrentBag<T>](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentbag-1?redirectedfrom=MSDN&view=netstandard-2.0)

Reply:
Changed to using `ConcurrentDictionary<,>`, but haven't tested it yet so hold off on merge pls